ref/Microsoft.Build.Tasks.Core/net/Microsoft.Build.Tasks.Core.cs

```diff
@@ -1217,6 +1217,8 @@ public sealed partial class Unzip : Microsoft.Build.Tasks.TaskExtension, Microso
         public Unzip() { }
         [Microsoft.Build.Framework.RequiredAttribute]
         public Microsoft.Build.Framework.ITaskItem DestinationFolder { get { throw null; } set { } }
+        public string Exclude { get { throw null; } set { } }
+        public string Include { get { throw null; } set { } }
```

Comment:
If we stick with regular expressions for this, we should change the names for `Exclude` and `Include` as they are "default" msbuild names. ie. items are added via Include and the patterns differ between them and regular expressions. A suggested name during PR review was `IncludePattern` & `ExcludePattern`

Reply:
This makes sense, the functionality differs indeed. I have pushed the change.

Reply:
@BenVillalobos, I thought we'd agreed globs were more MSBuild-y than regex?

Reply:
We did, I posted this while we were talking about it, hence the "If we stick with regular expressions." The current implementation is still regex, is this how we process includes and excludes normally?

Reply:
No; we use globs.

Reply:
So we're in agreement 🙂

@IvanLieckens the to-do list here would be:
- [ ] Rename ExcludePattern and IncludePattern back to Exclude and Include (sorry)
   - We wanted ExcludePattern if we were sticking with RegEx, but it turns out we don't want to use regex here
- [ ] Change the implementation to expect * parse a glob pattern

Reply:
@BenVillalobos ok, thank you for the clearing up. Is there a good example in MSBuild of how you'd like this to function I can use as a lead for the implementation? Just want to make sure that it feels native without any quirks.

Reply:
Check out `Expander.cs`, `ExpandIntoItemsLeaveEscaped` may have what you're looking for.

Reply:
`ItemSpec.cs` is also relevant here.

Reply:
Thank you @BenVillalobos , I studied Expander and ItemSpec but they were outside of reach for the Tasks assembly and I did not want to introduce dependencies so I leveraged FileMatcher to handle the normalization and verification of the globs and paths. It does not support property references due to this. I hope this matches with your expected behavior.

Reply:
[MSBuildGlob](https://github.com/dotnet/msbuild/blob/master/src/Build/Globbing/MSBuildGlob.cs) would have been great here, but unfortunately it's not visible by tasks. :(